File: 5 Modelo POO Javascript/src/com/tourmanager/ejemplo/ejemplo-uso.js
const {
  Cliente, Contrato, Estado, EstadoReserva, MetodoPago, PaqueteTuristico,
  Proveedor, Reserva, Rol, ServicioPaquete, TipoServicio, TipoTransporte,
  Transporte, Usuario, Venta
} = require('../modelo/modelo-agencia');

const administrador = new Rol(1, 'Administrador', 'Gestion total del sistema');
const usuario = new Usuario(1, 'Ana Torres', 'ana@tourmanager.com', '1001', administrador);
const cliente = new Cliente(1, 'Carlos', 'Perez', 'carlos@email.com');
const proveedor = new Proveedor(1, 'Aventuras Colombia', TipoServicio.TRANSPORTE);
proveedor.agregarContrato(new Contrato(1, 'CTR-001', new Date(Date.now() - 864000000), new Date(Date.now() + 180 * 86400000)));

const paquete = new PaqueteTuristico(1, 'Caribe inolvidable', 'Cartagena y alrededores', 1800000, new Date(Date.now() + 30 * 86400000), new Date(Date.now() + 35 * 86400000));
paquete.agregarServicio(new ServicioPaquete(1, TipoServicio.HOSPEDAJE, 'Hotel cinco noches', 700000));
paquete.asignarTransporte(new Transporte(1, TipoTransporte.AVION, proveedor, 180, 500000));

const reserva = new Reserva(1, cliente, paquete, 2);
reserva.usuarioGestor = usuario;
cliente.agregarReserva(reserva);
reserva.confirmar();

const venta = new Venta(1, reserva, reserva.total, MetodoPago.TRANSFERENCIA);
venta.registrarPago();

console.log(`Reserva ${reserva.id} - ${reserva.estado}`);
console.log(`Venta pagada: ${venta.estaPagada()}`);
console.log(`Proveedor con contrato vigente: ${proveedor.tieneContratoVigente()}`);
console.log(`Cliente: ${cliente.getNombreCompleto()}`);
console.log(`Estado activo del usuario: ${usuario.estado === Estado.ACTIVO}`);
console.log(`Estado esperado de reserva: ${EstadoReserva.CONFIRMADA}`);

File: 5 Modelo POO/src/com/tourmanager/ejemplo/EjemploUso.java
package com.tourmanager.ejemplo;

import com.tourmanager.modelo.ModeloAgencia.Cliente;
import com.tourmanager.modelo.ModeloAgencia.Contrato;
import com.tourmanager.modelo.ModeloAgencia.Estado;
import com.tourmanager.modelo.ModeloAgencia.EstadoReserva;
import com.tourmanager.modelo.ModeloAgencia.Itinerario;
import com.tourmanager.modelo.ModeloAgencia.MetodoPago;
import com.tourmanager.modelo.ModeloAgencia.PaqueteTuristico;
import com.tourmanager.modelo.ModeloAgencia.Proveedor;
import com.tourmanager.modelo.ModeloAgencia.Reserva;
import com.tourmanager.modelo.ModeloAgencia.Rol;
import com.tourmanager.modelo.ModeloAgencia.ServicioPaquete;
import com.tourmanager.modelo.ModeloAgencia.TipoServicio;
import com.tourmanager.modelo.ModeloAgencia.TipoTransporte;
import com.tourmanager.modelo.ModeloAgencia.Transporte;
import com.tourmanager.modelo.ModeloAgencia.Usuario;
import com.tourmanager.modelo.ModeloAgencia.Venta;

import java.math.BigDecimal;
import java.time.LocalDate;

public final class EjemploUso {
    private EjemploUso() { }

    public static void main(String[] args) {
        Rol administrador = new Rol(1L, "Administrador", "Gestion total del sistema");
        Usuario usuario = new Usuario(1L, "Ana Torres", "ana@tourmanager.com", "1001", administrador);
        Cliente cliente = new Cliente(1L, "Carlos", "Perez", "carlos@email.com");

        Proveedor proveedor = new Proveedor(1L, "Aventuras Colombia", TipoServicio.TRANSPORTE);
        proveedor.agregarContrato(new Contrato(1L, "CTR-001", LocalDate.now().minusDays(10), LocalDate.now().plusMonths(6)));

        PaqueteTuristico paquete = new PaqueteTuristico(
                1L, "Caribe inolvidable", "Cartagena y alrededores", new BigDecimal("1800000"),
                LocalDate.now().plusDays(30), LocalDate.now().plusDays(35));
        paquete.agregarServicio(new ServicioPaquete(1L, TipoServicio.HOSPEDAJE, "Hotel cinco noches", new BigDecimal("700000")));
        paquete.agregarItinerario(new Itinerario(1L, "Plan Caribe", "Actividades de playa y cultura"));
        paquete.asignarTransporte(new Transporte(1L, TipoTransporte.AVION, proveedor, 180, new BigDecimal("500000")));

        Reserva reserva = new Reserva(1L, cliente, paquete, 2);
        reserva.setUsuarioGestor(usuario);
        cliente.agregarReserva(reserva);
        reserva.confirmar();

        Venta venta = new Venta(1L, reserva, reserva.getTotal(), MetodoPago.TRANSFERENCIA);
        venta.registrarPago();

        System.out.println("Reserva " + reserva.getId() + " - " + reserva.getEstado());
        System.out.println("Venta pagada: " + venta.estaPagada());
        System.out.println("Proveedor con contrato vigente: " + proveedor.tieneContratoVigente());
        System.out.println("Cliente: " + cliente.getNombreCompleto());
        System.out.println("Estado activo del usuario: " + (usuario.getEstado() == Estado.ACTIVO));
        System.out.println("Estado esperado de reserva: " + EstadoReserva.CONFIRMADA);
    }
}
